### PR TITLE
Dls 8078 - added CYA guard, 

### PR DESCRIPTION
**Pull request by ChrisHillCap**
refactor of some stubs, refactor of some IT test names, added in dropping db before each it test so i had to amend some tests

### CODE REVIEW

File: it/controllers/ReturnSentControllerIntegrationSpec.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ class ReturnSentControllerIntegrationSpec extends Specifications with TestConfig
 //      val userAnswers = smallProducerDetaisPartialAnswers.success.value
 //      setUpData(userAnswers.copy(submitted = false))
 //      given
-//        .commonPrecondition
+//        .commonPrecondition(aSubscription)
 //
 //      WsTestClient.withClient { client =>
 //        val result1 = client.url(s"$baseUrl/return-sent")
```
Comment (RaminEsfandiari) [inline it/controllers/ReturnSentControllerIntegrationSpec.scala:44]: Can these be removed?

File: it/controllers/testSupport/TestConfiguration.scala [hunk 2]
```diff
@@ -148,7 +150,10 @@ trait TestConfiguration
   }
 
   override def beforeEach() = {
+    Await.result(mongo.collection.deleteMany(BsonDocument()).toFuture(),Duration(5,TimeUnit.SECONDS))
+    Await.result(sdilSessionCacheRepo.collection.deleteMany(BsonDocument()).toFuture(),Duration(5,TimeUnit.SECONDS))
```
Comment (RaminEsfandiari) [inline it/controllers/testSupport/TestConfiguration.scala:154]: Can these not be done in a non-blocking manner? If these are using 10 seconds  every time before each is called are they not adding a few minutes to the pipeline duration?

**Review by RaminEsfandiari — APPROVED** [review]
I'll approve and let you decide if the Await change on mongo stuff needs changing.

File: it/controllers/testSupport/TestConfiguration.scala (hunk 2 above)
Comment (ChrisHillCap) [reply it/controllers/testSupport/TestConfiguration.scala:154]: its non blocking, await waits for it to complete, duration is a timeout

File: it/controllers/ReturnSentControllerIntegrationSpec.scala (hunk 1 above)
Comment (ChrisHillCap) [reply it/controllers/ReturnSentControllerIntegrationSpec.scala:44]: this is christines pr. she will need it i guess

**Pull request merged**